src/Abstractions/NexusMods.Abstractions.NexusModsLibrary/Models/CollectionRevisionModFile.cs

```diff
@@ -0,0 +1,30 @@
+using NexusMods.Abstractions.MnemonicDB.Attributes;
+using NexusMods.MnemonicDB.Abstractions.Attributes;
+using NexusMods.MnemonicDB.Abstractions.Models;
+
+namespace NexusMods.Abstractions.NexusModsLibrary.Models;
+
+public partial class CollectionRevisionModFile : IModelDefinition
+{
+    private const string Namespace = "NexusMods.Library.CollectionRevisionModFile";
+    
+    /// <summary>
+    /// The Nexus, globally unique id identifying a specific file of a collection revision.
+    /// </summary>
+    public static readonly ULongAttribute FileId = new(Namespace, nameof(FileId)) { IsIndexed = true };
```

Comment:
This field is misleading.
It maps to the `Id` field of the `ModFiles` node in the GraphQL response; however the `ModFiles` node (and others) have a field called `FileId`.
Therefore, this field could very easily be mistaken for the `FileId` field present in the GraphQL API responses; when it is not the case.

------

**Potential Improvement**:

I think there's something better we can actually store here.
Storing the `Id` field doesn't do us much here, while it is a unique identifier, none of the GraphQL APIs actually support it as input.

What they do support however, is a field called `uid`.
The UID is a composite 8 byte key. The upper 4 bytes represent the item ID, and the lower 4 bytes represent the item group.

Example: [My branch for mod update cache stuff has a definition for Uid for a mod page.](https://github.com/Nexus-Mods/NexusMods.App/blob/71a131b273ca1600c23f6abc2c66f1df4d392cf0/src/Networking/NexusMods.Networking.ModUpdates/Structures/Uid.cs#L12) [I'll probably move this struct to main WebAPI project]

For files the definition is the same, the upper 4 bytes are the `FileId`, and the lower 4 bytes are `GameId`.
Since the API response gives you those fields back you can reconstruct the `uid`.

Which in turn, gives us more flexibility as we can use the V2 APIs to query these files (unlike the 'ID' field):

```graphql
query ModFiles {
    modFilesByUid(uids: ["14315126081750"]) {
        nodesCount
        nodesFilter
        totalCount
        nodes {
            category
            categoryId
            changelogText
            count
            date
            description
            fileId
            id
            manager
            modId
            name
            primary
            reportLink
            requirementsAlert
            scanned
            size
            sizeInBytes
            totalDownloads
            uCount
            uid
            uniqueDownloads
            uri
            version
        }
    }
}
```

Response:

```json
{
    "data": {
        "modFilesByUid": {
            "nodesCount": 1,
            "nodesFilter": null,
            "totalCount": 1,
            "nodes": [
                {
                    "category": "ARCHIVED",
                    "categoryId": "ARCHIVED",
                    "changelogText": [
                        "#<ModChangelog:0x00007fda4e6f3df0>"
                    ],
                    "count": 31009,
                    "date": 1722900467,
                    "description": "",
                    "fileId": 84182,
                    "id": "Z2lkOi8vYXBpL01vZEZpbGUvMTQzMTUxMjYwODE3NTA",
                    "manager": 0,
                    "modId": 4198,
                    "name": "ArchiveXL",
                    "primary": 0,
                    "reportLink": "https://www.virustotal.com/gui/file/cf5c14c6ad61839a6e2a69a026143bd134964028121ee29e6020ebc1081f5396/detection/f-cf5c14c6ad61839a6e2a69a026143bd134964028121ee29e6020ebc1081f5396-1722900543",
                    "requirementsAlert": 1,
                    "scanned": "VERIFIED",
                    "size": 774,
                    "sizeInBytes": "793064",
                    "totalDownloads": 31009,
                    "uCount": 26076,
                    "uid": "14315126081750",
                    "uniqueDownloads": 26076,
                    "uri": "ArchiveXL-4198-1-16-4-1722900467.zip",
                    "version": "1.16.4"
                }
            ]
        }
    }
}
```

etc.

(The UID input was reconstructed from `FileId: 84182`, `GameId: 3333`; from response in collection slug `iszwwe` revision `469`)

So I believe it's better to reconstruct `Uid` here; as this in turn allows us to make use of the V2 APIs for further querying (if we ever need it).

Since this is technically implementation detail, I did ask backend about this a while back when doing my own research for Reloaded3 stuff. They don't intend to change this, and in fact; it appears that it's now also sneakily documented in the GraphQL API site.

![image](https://github.com/user-attachments/assets/defc664c-88f2-44e0-b4cd-6cd6ec147ed4)